alchemiscale/storage/statestore.py

```diff
@@ -730,6 +745,96 @@ def set_strategy(
         """Set the compute Strategy for the given AlchemicalNetwork."""
         ...
 
+    def register_computeservice(
+        self, compute_service_registration: ComputeServiceRegistration
+    ):
+        """Register a ComputeServiceRegistration uniquely identifying a running
+        ComputeService.
+
+        A ComputeServiceRegistration node is used for CLAIMS relationships on
+        Tasks to avoid collisions in Task execution.
+
+        """
+
+        node = Node(
+            "ComputeServiceRegistration", **compute_service_registration.to_dict()
+        )
+
+        with self.transaction() as tx:
+            tx.merge(
```

Comment:
Does this merge allow you to attempt to register the same `compute_service` more than once? IMO should be an error if you try and overwrite and existing service or at least give some feedback if the query did nothing ie found an already existing service. I could be missing something that prevents this though...

Reply:
Good question!

We're using `py2neo`'s `Transaction.merge` here: https://py2neo.org/2021.1/workflow.html?highlight=merge#py2neo.Transaction.merge

It won't create a `ComputeServiceRegistration` with the same `identifier`, and for additional safety on this we also set a uniqueness constraint for `ComputeServiceRegistration`s on their `identifier` (see above). That being said, no feedback will be given if the same `ComputeServiceRegistration` is trying to be used twice; I don't think this will be a problem in practice, but you raise a good point that we also don't want this happening.

I can change this to use `tx.create` instead, which will raise an exception if the object exists already due to the constraint we have set above.